cmd/oras/root/manifest/index/update.go

```diff
@@ -0,0 +1,243 @@
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package index
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/opencontainers/go-digest"
+	"github.com/opencontainers/image-spec/specs-go"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/spf13/cobra"
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/content"
+	"oras.land/oras/cmd/oras/internal/argument"
+	"oras.land/oras/cmd/oras/internal/command"
+	"oras.land/oras/cmd/oras/internal/display/status"
+	oerrors "oras.land/oras/cmd/oras/internal/errors"
+	"oras.land/oras/cmd/oras/internal/option"
+	"oras.land/oras/cmd/oras/internal/output"
+	"oras.land/oras/internal/contentutil"
+	"oras.land/oras/internal/descriptor"
+)
+
+type updateOptions struct {
+	option.Common
+	option.Target
+
+	addArguments    []string
+	mergeArguments  []string
+	removeArguments []string
+	tags            []string
+}
+
+func updateCmd() *cobra.Command {
+	var opts updateOptions
+	cmd := &cobra.Command{
+		Use:   "update <name>{:<tag>|@<digest>} {--add/--merge/--remove} {<tag>|<digest>} [...]",
+		Short: "[Experimental] Update and push an image index",
+		Long: `[Experimental] Update and push an image index. All manifests should be in the same repository
+		
+Example - add one manifest and remove two manifests from an index tagged 'latest'. The tag will point to the updated index:
+  oras manifest index update localhost:5000/hello:latest --add linux-amd64 --remove sha256:xxx
+
+Example - update an index by specifying its digest:
+  oras manifest index update localhost:5000/hello@sha256:xxx --add sha256:xxx --remove linux-arm64
+
+Example - remove a manifest and merge manifests from the index tagged as 'index01':
+  oras manifest index update localhost:5000/hello:latest --remove sha256:xxx --merge index01
+
+Example - update an index and tag the updated index:
+  oras manifest index update localhost:5000/hello@sha256:xxx --remove sha256:xxx --tag "latest" --tag "v2.1.0"
+  `,
+		Args: oerrors.CheckArgs(argument.AtLeast(1), "the destination index to update"),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			opts.RawReference = args[0]
+			if err := option.Parse(cmd, &opts); err != nil {
+				return err
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return updateIndex(cmd, opts)
+		},
+	}
+	option.ApplyFlags(&opts, cmd.Flags())
+	cmd.Flags().StringArrayVarP(&opts.addArguments, "add", "", nil, "add manifests to the index")
+	cmd.Flags().StringArrayVarP(&opts.mergeArguments, "merge", "", nil, "merge the manifests of another index")
```

Comment:
"merge" sounds a bit confusing to me. Does it mean adding the list of manifests to the target index from another index?

Reply:
Yes exactly.

Reply:
Can we make it clear in the description?